lib/model/AjvValidator.js

```diff
@@ -244,12 +244,33 @@ function objectHasDefaults(obj) {
   return false;
 }
 
+function deepRemoveRequired(jsonSchemaProperties) {
+  if (jsonSchemaProperties === undefined) {
+    return;
+  }
+
+  if (jsonSchemaProperties.required !== undefined) {
+    delete jsonSchemaProperties.required;
+  }
+
+  if (jsonSchemaProperties.type !== 'object' || jsonSchemaProperties.properties === undefined) {
+    return;
+  }
+
+  for (const key of Object.keys(jsonSchemaProperties.properties)) {
+    deepRemoveRequired(jsonSchemaProperties.properties[key]);
+  }
+}
+
 function jsonSchemaWithoutRequired(jsonSchema) {
   const subSchemaProps = ['anyOf', 'oneOf', 'allOf', 'not', 'then', 'else'];
 
+  const clonedSchema = Object.assign(jsonSchema, {});
```

Comment:
You only clone the outermost level of the object, but modify all levels in the `deepRemoveRequired`. That will end up modifying the original schema.

Reply:
Instead of using deepRemoveRequired, which mutates the object, you should recursively apply `subSchemaWithoutRequired` for all properties of type `object` or `array` with objects in them.

Reply:
Are you opposed to just doing a cloneDeep at the top of compilePatchValidator() ? This way we ensure the original object never gets mutated and it could simplify some stuff downsteam as a result.